### DISCUSSION TITLE
Display only today data

### DISCUSSION THREAD

**Discussion by vaskoZGU** [Budibase/budibase#11314] [Help] [answered]
Hello,

I use external MySQL DB. 
I need help displaying data for today's date only. 
Assuming today is 9 Nov 2022 (2022-11-09) my goal is to display only first 2 rows in table for the app users.
![image](https://github.com/Budibase/budibase/assets/117993754/8fb56c2e-6c0e-4fa7-9b09-d5aae7ef72c8)

Thank you.

**Comment by ConorWebb96** [comment]
Hey @vaskoZGU,

From the looks of the above screenshot it seems to be the sample data provided via using the internal Budibase db, not an external MySQL DB.

Theres a few different ways you can achieve this. Are you wanting this to occur on load or are you wanting the users to be able to change this in the frontend? Below are a few quick examples showing how you can do this;

You can use the dynamic filter component and write the query within the frontend using it.
**Componet tree**
<img width="100%" alt="Screenshot 2023-07-21 at 11 52 19" src="https://github.com/Budibase/budibase/assets/126772285/ca747446-a8cc-422d-a0eb-25ea7d090ef0">

**Output**
<img width="1259" alt="Screenshot 2023-07-21 at 11 52 19" src="https://github.com/Budibase/budibase/assets/126772285/f4085bb7-d863-4c38-8abf-62aac8105d0e">

Or you can filter directly on the `Data provider`
<img width="1438" alt="Screenshot 2023-07-21 at 12 00 34" src="https://github.com/Budibase/budibase/assets/126772285/ad51885d-fc51-4622-bd08-c74e7f3ce9b8">

You can even use a date picker field to allow more control, instead of hardcoding the filter like the above you'd supply the form fields value as the value checked against.

I hope this makes sense and is helpful.

**Reply by vaskoZGU** [reply]
Hi @ConorWebb96,


Thank you for your reply.
Actually I use the sample data for testing, but the real backend is MariaDB.

Users add data daily via a form.
They should not see any data from the past in the table, only data added during the current day.
The next day they should start with an empty table (filtered).

If I create a filter in the data provider as in your last screenshot, the user will see the past data from 2022-11-01 (assuming today is 2022-11-09). Am I right?

**Reply by ConorWebb96** [reply, marked as answer]
Hey @vaskoZGU,

If you want only records of the current day I'd recommend adding the below JS inside that binding were that date is displayed in the last screenshot.

```
const currentDate = new Date();
currentDate.setHours(0, 0, 0, 0);

return currentDate;
```
This will always default it to the current date at 00:00, meaning your data will be filtered by dates on a specific day.

I hope this makes sense.

**Reply by vaskoZGU** [reply]
Thank you @ConorWebb96 

This is what I was looking for.
